scala-package/core/src/main/scala/org/apache/mxnet/module/BaseModule.scala

```diff
@@ -549,6 +560,25 @@ abstract class BaseModule {
            forceRebind: Boolean = false, sharedModule: Option[BaseModule] = None,
            gradReq: String = "write"): Unit
 
+
+ /**
+  * Bind the symbols to construct executors.
+  * This is necessary before one can perform computation with the module.
+  * @param forTraining Default is `True`. Whether the executors should be bind for training.
+  * @param inputsNeedGrad  Default is `False`.
+  *                        Whether the gradients to the input data need to be computed.
+  *                        Typically this is not needed.
+  *                        But this might be needed when implementing composition of modules.
+  * @param forceRebind Default is `False`. This function does nothing
+  *                    if the executors are already binded. But with this `True`,
+  *                    the executors will be forced to rebind.
+  * @param dataShape Typically is `DataIter.provideData`.
+  */
+  @varargs def bind(forTraining: Boolean, inputsNeedGrad: Boolean,
```

Comment:
doesn't this bind need to support labelShapes/ sharedModule for bucketing?

Reply:
I don't think this need to be experiment. experiment is for something that is unstable, for example, I think technically type-safe APIs should be marked as experiment. it just use the default args for labelShapes/ sharedModule, to make it easier to users, similar as these bind: https://github.com/apache/incubator-mxnet/blob/master/scala-package/core/src/main/scala/org/apache/mxnet/Symbol.scala#L691-L709 do they need to be marked experiment as well?